handbook/engineering/documentation.md

```diff
@@ -1,6 +1,8 @@
-# Sourcegraph documentation guidelines
+# Sourcegraph engineering documentation
```

Comment:
Let's call this product documentation. Can we rename the file also to `product_documentation.md`?

```suggestion
# Product documentation
```

Reply:
Yep, will rename.